backend/services/implementations/authService.ts

```
@@ -160,6 +160,23 @@ class AuthService implements IAuthService {
     }
   }
 
+  async resetPasswordCode(email: string): Promise<string> {
+    try {
+      const resetLink = await firebaseAdmin
+        .auth()
+        .generatePasswordResetLink(email);
+      const regex = /(?<=&oobCode=)(.*)(?=&apiKey=)/gm;
+      const matches = resetLink.match(regex);
+      if (matches) return matches[0];
+      return "";
```

should we be throwing an error here instead of an empty string? & same for the functions below